Rewrite datastreamr

kkirov opened the pull request:
This rewrite update the request patterns to be in line with our API best practices and be consistent with other packages.

kkirov committed:
feat: rewrite the datastreamr based on the Javascript implementation

kkirov committed:
docs: update man and Readme

kkirov committed:
docs: remove or

kkirov committed:
refactor: paths

kkirov committed:
fix: loading options for packages

kkirov committed:
feat: convert results to data.frame

kkirov committed:
fix: convert to dataframe

kkirov committed:
fix: pagination

kkirov committed:
Merge remote-tracking branch 'origin/main' into rewrite-datastreamr

Commit:
Split functions into 3 files (util, fecth, datastreamr) for easier maintinance.
Fixed some general syntax issues (still more to go).
Made utility and fetch functions internal.
Removed library calls, and function calls in main script files.
Fixed examples so that they actually run.
Added the \dontrun{} tag to examples since they rely on a valid API key.
Added package names with :: to remove import requirments.

Commit:
Updated NAMESPACE so it can me maintained automatically.

Commit:
Updated DESCRIPTION file removing unused dependencies, and moving packages needed only for testing to SUGGESTS.

Commit:
updated all help files. Included the 'internal' keywords tag for internal functions, and `dontrun` tags to all examples since they need a valid api key.

Commit:
Updated testing scripts.
Moved functions out of the "testing" files and into "helper" files.
Fixed an issue with 'local_mocked_bindings' not being setup correctly.

Commit:
updated installed packages to reflect what is needed by datastreamr.
Added additional r script files to copied files.

Commit:
Updated DESCRIPTION to modern Authors@R approach for more detailed metadata. Note I've listed Patrick as 'cre' role, designating myself as maintained. R does not distinguish creator from maintainer, and only one maintainer may be specified on a package, despite Kiril being mostly responsible for the creation of the package I think it makes sense for DataStream to be listed as the maintainer.

Commit:
Updated MIT License

Commit:
Added quite Option
Setting options(datastream_quite = TRUE) will print the api calls to the console for debugging.

Commit:
Updated quite option
Cleaned syntax

lorilogan reviewed (commented):
Patrick did you mean "quiet"?

Commit:
fixed typo

Commit:
replaced httr with httr2
httr::content() was giving warnings about when as = "parse" and type = "application/json". httr 2 superseeds httr, and should be more stable. Updated testing files accordingly.

p-schaefer reviewed (approved):
Still some syntax that could be cleaned, but code is in good shape. I haven't tested the docker component.
